### PR TITLE
fix: alter expensify-datepicker class

### PR DESCRIPTION
**Pull request by shonsirsha**
### Details
When date picker is visible (normally on desktop app & desktop web), date picker icon should be centred (vertically) and its touchable/clickable area should be increased.
### Fixed Issues

$ [12957](https://github.com/Expensify/App/issues/12957)   
PROPOSAL: [GH_LINK_ISSUE (COMMENT)](https://github.com/Expensify/App/issues/12957#issuecomment-1330084343) - a super slight modification of it.

### Tests
1. Go to any view with a date picker, for example: Workspace -> Connect Bank Account -> Connect Manual -> Company Step -> Incorporation date
2. If Icon is visible, it should be in the centre (vertically)
3. If Icon is visible, it should still be clickable even when not clicking **exactly** on it (has a sufficient surface / touchable / clickable area). Upon successful click, it opens the datepicker dialogue.

- [x] Verify that no errors appear in the JS console

### Offline tests
1. Turn network to offline 
2. Go to any view with a date picker that's available offline
3. If Icon is visible, it should be in the centre (vertically)
4. If Icon is visible, it should still be clickable even when not clicking **exactly** on it (has a sufficient surface / touchable / clickable area). Upon successful click, it opens the datepicker dialogue.

### QA Steps
1. Go to any view with a date picker, for example: Workspace -> Connect Bank Account -> Connect Manual -> Company Step -> Incorporation date
2. If Icon is visible, it should be in the centre (vertically)
3. Icon is still clickable which opens the date dialogue even when not clicking **exactly** on it (has a sufficient surface / touchable / clickable area).

- [ ] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] iOS / native
    - [x] Android / native
    - [x] iOS / Safari
    - [x] Android / Chrome
    - [x] MacOS / Chrome
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

https://user-images.githubusercontent.com/28676406/205505710-1de96149-4ed2-4f4a-80b6-bf861cea70b0.mov

</details>

<details>
<summary>Mobile Web - Chrome</summary>

https://user-images.githubusercontent.com/28676406/205507151-70bcbe07-53d9-4acd-9119-7007d194d65f.mov

</details>

<details>
<summary>Mobile Web - Safari</summary>

https://user-images.githubusercontent.com/28676406/205505715-d42e614e-2cf3-4e9d-bd90-a6aab7c065f2.MP4

</details>

<details>
<summary>Desktop</summary>

https://user-images.githubusercontent.com/28676406/205505891-ca3dbce2-b2e8-4334-871e-ef5e85009320.mov

</details>

<details>
<summary>iOS</summary>

https://user-images.githubusercontent.com/28676406/205551282-1ee134b7-e359-4b36-aa11-3d0cff720e2d.mov

</details>

<details>
<summary>Android</summary>

https://user-images.githubusercontent.com/28676406/205507339-0a2efed2-5149-4dc0-850d-6650f6898da8.mov

</details>

### CODE REVIEW

**Review by sobitneupane — COMMENTED** [review]
Screen Recording is missing for IOS in PR body. Can you please update it to include screen recording for IOS as well.

**Comment by shonsirsha** [comment]
@sobitneupane Thanks for the comments, changes addressed. PTAL 🙏 

Screen Shot 2022-12-05 at 04 42 18 

I also altered the font size (calendar icon's size) to be the same as the other icons - as it was around 5-6px smaller before. 


If we keep the icon smaller than the others, combined with the padding that we have right now (`18px 9px`, which made it align nicely to the right) it's gonna look odd (too much whitespace on left side). 

Thus, I increased the size of the calendar icon - so all icons have a uniform size. Thanks.

**Comment by sobitneupane** [comment]
@shonsirsha There are different factors that determine the font size of other elements. So, comparing in one platform and using a value is not the right option. Let's not change the font size.

**Comment by shonsirsha** [comment]
> @shonsirsha There are different factors that determine the font size of elements. So, comparing in one platform and using a value is not the right option. Let's not change the font size.

Hi, sure!  🙂 reverted

Screen Shot 2022-12-05 at 06 48 58

**Comment by shonsirsha** [comment]
Oops, I re-requested review and it automatically removed @deetergp from the reviewers list. Please feel free to add yourself in again if it's necessary. Thanks!

**Comment by shonsirsha** [comment]
Hey, sorry for spamming but do we have any updates here? 🙂 Please lmk if I'm missing something, thanks.

File: src/components/DatePicker/styles.css [hunk 1]
```diff
@@ -6,10 +6,13 @@
  */
 
 .expensify-datepicker::-webkit-calendar-picker-indicator {
-    position: absolute;
-    right: 12px;
-    top: 13px;
     color-scheme: dark;
+    cursor: pointer;
+    padding: 18px 9px;
```
Comment (sobitneupane) [inline src/components/DatePicker/styles.css:11]:
```suggestion
    padding: 18px 13px;
```

Comment (shonsirsha) [reply src/components/DatePicker/styles.css]: No probs! Changes made 🙂

**Comment by shonsirsha** [comment]
Hey 👋 , I believe we are good here..? Any possibility to get an approval or any updates? 

Sorry, it's my first PR to this repo - is something else needs to be done from my side? 🙂

**Review by deetergp — APPROVED** [review]
Changes look good, but you need to sign the CLA @shonsirsha and @sobitneupane needs to compete the reviewer checklist in order for the tests to pass so that the PR can be merged.

**Comment by deetergp** [comment]
## Reviewer Checklist

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] iOS / native
    - [ ] Android / native
    - [ ] iOS / Safari
    - [ ] Android / Chrome
    - [ ] MacOS / Chrome
    - [ ] MacOS / Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [ ] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>



</details>

<details>
<summary>Mobile Web - Chrome</summary>



</details>

<details>
<summary>Mobile Web - Safari</summary>



</details>

<details>
<summary>Desktop</summary>



</details>

<details>
<summary>iOS</summary>



</details>

<details>
<summary>Android</summary>



</details>

**Comment by shonsirsha** [comment]
_**I have read the CLA Document and I hereby sign the CLA**_

**Comment by shonsirsha** [comment]
Thanks a lot @deetergp! Seems like like I have [sucessfully signed the CLA](https://github.com/Expensify/CLA/commit/ca2cf82aedb48ede98488c649549bd29b9533d33). 

The check in this PR is still failing though. How does one re-run this check?

**Review by sobitneupane — APPROVED** [review]
### Screenshots/Videos
<details>
<summary>Web</summary>
Screenshot 2022-12-07 at 09 04 50
Screenshot 2022-12-07 at 09 11 52
Screenshot 2022-12-07 at 09 15 38

</details>

<details>
<summary>Mobile Web - Chrome</summary>




</details>

<details>
<summary>Mobile Web - Safari</summary>



</details>

<details>
<summary>Desktop</summary>

Screenshot 2022-12-07 at 09 07 27
Screenshot 2022-12-07 at 09 17 16



</details>

<details>
<summary>iOS</summary>




</details>

<details>
<summary>Android</summary>



</details>

**Comment by sobitneupane** [comment]
Changes look good and test well.

🎀👀🎀 C+ reviewed

cc: @deetergp @grgia

**Pull request merged**